src/lib/routes/admin-api/project/variants.ts

```diff
@@ -80,6 +83,10 @@ export default class VariantsController extends Controller {
             handler: this.patchVariants,
             middleware: [
                 openApiService.validPath({
+                    summary:
+                        'This will apply the same patch to variants in all environments.',
+                    description:
+                        'This method is not atomic which is the main reason why it is not recommended to use.',
```

Comment:
What do we recommend that you do instead of using this endpoint?

```suggestion
                        `Apply the provided patch to the specified feature's variants. The patch conform to [some format] @gastonfournier 
                        
                        ⚠️ **Warning**: This method is not atomic. If something fails in the middle of applying the patch, you can be left with a half-applied patch. We recommend that you instead [do what] ... @gastonfournier?`,
```

Reply:
> Apply the provided patch to the specified feature's variants. The patch conforms to [some format] @gastonfournier 

Not sure what you want to say here. Shouldn't our open API schema be enough? 

Internally, we use an implementation of JSON-Patch [(RFC6902)](http://tools.ietf.org/html/rfc6902), but we wrap it with some additional data. 

> ⚠️ **Warning**: This method is not atomic. If something fails in the middle of applying the patch, you can be left with a half-applied patch. We recommend that you instead [do what] ... @gastonfournier?

In this case, we recommend you use the same method by environment. The operation is: https://docs.getunleash.io/reference/api/unleash/patch-environments-feature-variants

Reply:
> Not sure what you want to say here. Shouldn't our open API schema be enough?

In theory, yes! I had a look now, and I think we've improved the payload for patch schemas. I think it used to be just any object (`{}`). However, the patch schema format is still pretty inscrutable if you haven't used it before (I always need to look it up), so it would be nice to at least have some examples and have it documented.

... buuut that's not the place of this PR, so I think we can leave that as is.

> Internally, we use an implementation of JSON-Patch (RFC6902), but we wrap it with some additional data.

What kind of additional wrapping do we do? Would doing something like this be appropriate?

As for the suggestion, then:

```suggestion
                        `Apply the provided patch to the specified feature's variants. The patch object should conform to the [JSON-patch format (RFC 6902)](https://www.rfc-editor.org/rfc/rfc6902).
                        
                        ⚠️ **Warning**: This method is not atomic. If something fails in the middle of applying the patch, you can be left with a half-applied patch. We recommend that you instead [patch variants on a per-environment basis](/docs/reference/api/unleash/patch-environments-feature-variants.api.mdx), which **is** an atomic operation.`,
```

Reply:
It is inscrutable for me as well, I think it's only helpful if you generate it from a diff (which I believe that's what we do in the UI). 

> What kind of additional wrapping do we do? Would doing something like this be appropriate?

To be precise the objects in the body array of https://docs.getunleash.io/reference/api/unleash/patch-environments-feature-variants is what should conform to the RFC spec

Reply:
Sooo ... this?

```suggestion
                        `Apply a list of patches patch to the specified feature's variants. The patch objects should conform to the [JSON-patch format (RFC 6902)](https://www.rfc-editor.org/rfc/rfc6902).
                        
                        ⚠️ **Warning**: This method is not atomic. If something fails in the middle of applying the patch, you can be left with a half-applied patch. We recommend that you instead [patch variants on a per-environment basis](/docs/reference/api/unleash/patch-environments-feature-variants.api.mdx), which **is** an atomic operation.`,
```